scripts/llm/gpt_train.py

```diff
@@ -159,21 +165,20 @@ def _read_chat_template(template_path: str):
         chat_template = _read_chat_template(args.chat_template_path)
         tokenizer = get_tokenizer(args.tokenizer, chat_template=chat_template)
         if '{% generation %}' not in tokenizer.tokenizer.chat_template:
-            if not args.chat_template_path:
-                raise ValueError(
-                    "Tokenizer does not contain the '{% generation %}' keyword. Please provide a chat template path using --chat-template-path."
-                )
-            raise ValueError(
-                "Please ensure the chat template includes a '{% generation %}' keyword for proper assistant mask during training. See https://github.com/huggingface/transformers/pull/30650"
+            logging.warning(
+                "Tokenizer does not contain the '{% generation %}' keyword. Please provide a chat template path using --chat-template-path that includes the '{% generation %}' keyword for proper assistant mask during training. See https://github.com/huggingface/transformers/pull/30650."
             )
+        ps = PackedSequenceSpecs(packed_sequence_size=args.seq_length)
         data = ChatDataModule(
             dataset_root=args.data_paths[0],
             seq_length=args.seq_length,
             tokenizer=tokenizer,
             global_batch_size=args.gbs,
             micro_batch_size=args.mbs,
             use_hf_tokenizer_chat_template=True,
+            packed_sequence_specs=ps if args.pack_sequence else None,
         )
+        data.prepare_data()  # sequence packing
```

Comment:
prepare_data shouldn't need to be called explicitly. PTL will call it on a single process